### DISCUSSION TITLE
How to know when a dynamically loaded component has been rendered

### DISCUSSION THREAD

**Discussion by acelaya** [vercel/next.js#21553] [Help]
I have a page where I dynamically render MDX components using Next.js' `dynamic` helper. That works fine:

```ts
import dynamic from 'next/dynamic';

export default function MyPage() {
    const path = determinePathSomehow()
    const Content = dynamic(() => import(path))

    return (
        <div>
            <Content />
        </div>
    )
}
```

All good here, however, my MDX files include hash links for every title, and since the content is dynamically loaded, the browser does not scroll to it when the url includes the hash (I'm not sure if this is always the case. It could work depending on how fast it loads).

As a workaround I have done this:

```ts
import dynamic from 'next/dynamic';

export default function MyPage() {
    const path = dynamicallyGetPathSomehow()
    const Content = dynamic(() => import(path).then((result) => {
        if (process.browser && window.location.hash) {
            const { hash } = window.location;

            setTimeout(() => {
                document.getElementById(hash.substr(1))?.scrollIntoView();
            }, 10);
        }

        return result
    }))

    return (
        <div>
            <Content />
        </div>
    )
}
```

This kind of "solves" the problem, but it's quite hacky. Ideally, I would like to be able to have some hook (`useEffect`?) where I can have this logic, the problem is that useEffect's callback runs when the main component renders, without waiting for the dynamic import.

Is there any way to run custom code once you know the dynamic content has finished loading? I have tried without the `setTimeout`, but in there, you only know that the component was loaded, but not rendered, so it also doesn't work.

**Comment by warrenalphonso** [comment]
Same question! There's gotta be a simple way to get `useEffect` to run after a dynamic import renders.
